demo/d2l-filter-dropdown/d2l-filter-dropdown.html

```diff
@@ -47,19 +47,66 @@ <h3>Basic d2l-filter-dropdown demo</h3>
 		</div>
 	</body>
 	<script type="module">
+		function generateOptions(i) {
+			const options = [];
+			for (let j = 0; j < 20; j++) {
+				options.push({key: `option${i}-${j}`, title: `Option ${i}-${j}`, selected: j === i});
+			}
+			return options;
+		}
+		function generateCategory(i) {
+			const options = generateOptions(i);
+			return {
+				key: 'Cat' + i,
+				title: 'Cat ' + i,
+				numSelected: options.filter(o => o.selected).length,
+				options: options
+			}
+		}
+
+		function search(search, category, data) {
+			if (search === '') {
+				return data[category].options;
+			}
+
+			return data[category].options.filter(opt => opt.title.indexOf(search) >= 0);
+		}
 		window.requestAnimationFrame(function() {
-			var filter = document.querySelector('d2l-filter-dropdown.with-search');
-			for (var i = 0; i < 6; i++) {
-				filter.addFilterCategory('Cat' + i, 'Cat ' + i, i === 0);
-				for (var j = 0; j < 20; j++) {
-					filter.addFilterOption('Cat' + i, `option${i}-${j}`, `Option ${i}-${j}`, j === i);
-				}
+			const filter = document.querySelector('d2l-filter-dropdown.with-search');
+			const data = {};
+			// Generate mirror of data
+			for (let i = 0; i < 6; i++) {
+				data['Cat' + i] = generateCategory(i);
 			}
-			filter = document.querySelector('d2l-filter-dropdown.no-search');
-			for (i = 0; i < 4; i++) {
-				filter.addFilterCategory('NCat' + i, 'NCat ' + i, i === 0);
-				for (j = 0; j <= i; j++) {
-					filter.addFilterOption('NCat' + i, `Noption${i}-${j}`, `NOption ${i}-${j}`, j === i);
+
+			// Set up filter with our generated data
+			Object.keys(data).forEach(categoryKey => {
```

Comment:
The diff in this file is me adding back search functionality to the demo. I am a bit worried that developers might think that this will come with search out-of-the-box. What are your thoughts?

Reply:
Yeah, I think I'd just log the filter-specific search event to the console, similar to what the `d2l-search-input`'s demo does: https://github.com/BrightspaceUI/inputs/blob/master/demo/d2l-input-search.html#L89.  This is what we do in most UI demos when the consumer is in charge of listening to an event and doing something

Reply:
This would also only work until I did that refactor we discussed offline

Reply:
I'm also seeing weird behaviour if you try to press the Clear button while you have a search applied, then remove the search.  Old applied option state is coming back.  I'd just go with the event!

Reply:
> I'm also seeing weird behaviour if you try to press the Clear button while you have a search applied, then remove the search

oh yeah I forgot to hook into the clear button!

> This is what we do in most UI demos when the consumer is in charge of listening to an event and doing something

Bugs aside, I think there is some value in having search work in the demo (for example to see a loading behaviour). At the very least it provides an example of how to implement search. Since I'm already basically done I would lean towards keeping it in. Does that work for you?

Reply:
Yeah that's fine, it's already written and seems like you know the bug fix.  I'm just secretly hoping no one does implement it this way before the refactor (other than the HM one)

Reply:
- [ ] fix clear/search interaction in demo